### ISSUE TITLE
[dataframes] `!=` comparison on columns produces results with error

### ISSUE THREAD

**Issue by maxim-uvarov** [nushell/nushell#11611]
### Describe the bug

`!=` comparison on dataframes works incorrectly, while `==` works ok.

### How to reproduce

```nushell
> let $df = ([[c]; [1] [2]] | dfr into-df )
> $df.c != $df.c
╭─#─┬neq_c_c╮
│ 0 │ true  │
│ 1 │ true  │
╰───┴───────╯
```

### Expected behavior

I expect this comparison to work.

### Configuration

| key                | value                                                |
| ------------------ | ---------------------------------------------------- |
| version            | 0.89.0                                               |
| branch             |                                                      |
| commit_hash        |                                                      |
| build_os           | macos-aarch64                                        |
| build_target       | aarch64-apple-darwin                                 |
| rust_version       | rustc 1.74.1 (a28077b28 2023-12-04)                  |
| rust_channel       | stable-aarch64-apple-darwin                          |
| cargo_version      | cargo 1.74.1 (ecb9851af 2023-10-18)                  |
| build_time         | 2024-01-10 08:30:05 +00:00                           |
| build_rust_channel | release                                              |
| allocator          | mimalloc                                             |
| features           | dataframe, default, extra, sqlite, trash, which, zip |
| installed_plugins  |                                                      |

### Additional context

as a clumsy workaround one can use:
`$df.c == $df.c | dfr into-df | dfr not`

Meanwhile, `dfr into-df` should not be necessary here, but otherwise, it just won't work.
I guess it is anohter bug:(

**Comment by fdncred** [comment]
My guess is because these all call `Series::equal` https://github.com/nushell/nushell/blob/4c5a8c1804885d368bbb0e5413c1170a81526983/crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/between_values.rs#L80-L109

**Comment by ayax79** [comment]
> My guess is because these all call `Series::equal`
Yes, that was it.. I'll have a pull request once I have some tests cases written.
